pkg/model/bootstrapscript.go

```diff
@@ -88,3 +94,67 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 	}
 	return fi.WrapResource(templateResource), nil
 }
+
+func (b *BootstrapScript) createProxyEnv(ps *kops.EgressProxySpec) string {
+	var buffer bytes.Buffer
+
+	if ps != nil && ps.HTTPProxy.Host != "" {
+		var httpProxyUrl string
+
+		// TODO double check that all the code does this
+		// TODO move this into a validate so we can enforce the string syntax
+		if !strings.HasPrefix(httpProxyUrl, "http://") {
+			httpProxyUrl = "http://"
```

Comment:
httpProxyUrl is always going to be empty at this point in the code, we created the variable a few lines above...